code/modules/reagents/chemistry/reagents/other_reagents.dm

```diff
@@ -1136,6 +1175,89 @@
 		M.confused = min(M.confused + 2, 5)
 	..()
 
+/datum/reagent/carbon_monoxide
+	name = "Carbon Monoxide"
+	description = "A highly dangerous gas for sapients."
+	reagent_state = GAS
+	metabolization_rate = REAGENTS_METABOLISM
+	color = "#96898c"
+	var/accumilation
+
+/datum/reagent/carbon_monoxide/on_mob_life(mob/living/carbon/victim)
+	if(holder.has_reagent(/datum/reagent/oxygen))
+		holder.remove_reagent(/datum/reagent/carbon_monoxide, 2*REM)
+		accumilation = accumilation/4
+
+	accumilation += volume
+	switch(accumilation)
+		if(10 to 50)
+			to_chat(src, "<span class='warning'>You feel dizzy.</span>")
```

Comment:
also this is going to fire every mob life check? give it a cooldown/timer or something